client/webserver/site/src/js/forms.ts

```diff
@@ -1124,26 +1124,39 @@ export class WalletWaitForm {
       Doc.hide(page.syncSpinner)
       Doc.show(page.syncUncheck)
     }
-    page.progress.textContent = String(Math.round(prog * 100))
-
-    // The remaining time estimate must be based on more than one progress
-    // report. We'll cache up to the last 20 and look at the difference between
-    // the first and last to make the estimate.
+    page.progress.textContent = String((prog * 100).toFixed(1))
+
+    if (prog >= 0.999) {
+      Doc.show(page.syncRemainBox)
+      // The final stage of wallet sync process can take a while (it might hang
+      // at 99.9% for many minutes, indexing addresses for example), the simplest
+      // way to handle it is to keep displaying constant rough estimate until
+      // sync is finished.
+      page.syncRemain.textContent = '~1 m'
+      return
+    }
```

Comment:
Might also create a separate message for **>= 99.9% case** (instead of displaying `~1 m remaining` like I'm currently doing here), but that would require more code/translations, seems like not worth the effort ?

Or maybe just not show this estimate altogether for **>= 99.9% case** since it won't always be accurate ?

Reply:
Tough call.  "~1 m" isn't terrible, and it's probably ok not to require translation.  Showing nothing isn't bad either IMO.  Something like "finishing up" could be best, but it's really not that necessary and its more to translate. So, meh

Reply:
IMO, if with will help with passing a more correct message to users, another translatable wording won't hurt. If it hangs there for a much longer time ~1 m is still confusing. It is 1 minute, a month? 🤔☺️ And even if users can tell what 1 m is but we are not done in that time frame, it could become a thing of concern.

Reply:
Changed to "finishing up":
<img width="256" alt="image" src="https://user-images.githubusercontent.com/112318969/219337178-81a0cdfd-5698-43bf-ac7c-f0e46c419b31.png">